File: crates/flowrt-codegen/tests/golden/bounded_operation_iox2_cpp/expected/app/stubs/cpp/navigator.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Navigator final : public flowrt_app::NavigatorInterface {
public:
    flowrt::OperationHandlerResult<flowrt_app::PlanResult> on_plan_operation(
        const flowrt_app::PlanGoal& goal,
        flowrt::OperationCancelToken cancel,
        flowrt::OperationProgressPublisher<flowrt_app::PlanFeedback>& progress) override {
        (void)goal;
        (void)cancel;
        (void)progress;
        return flowrt::OperationHandlerResult<flowrt_app::PlanResult>::failed();
    }

    flowrt::Status on_tick() override {
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Navigator>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/bounded_operation_iox2_rust/expected/app/stubs/rust/navigator.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Navigator;

impl flowrt_app::components::Navigator for Navigator {
    fn on_plan_operation(
        &mut self,
        goal: &flowrt_app::messages::PlanGoal,
        cancel: flowrt::OperationCancelToken,
        progress: &mut flowrt::OperationProgressPublisher<flowrt_app::messages::PlanFeedback>,
    ) -> flowrt::OperationHandlerResult<flowrt_app::messages::PlanResult> {
        let _ = (goal, cancel, progress);
        flowrt::OperationHandlerResult::failed()
    }

    fn on_tick(&mut self) -> flowrt::Status {
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Navigator::default()))
}
